Modules/LearningSequence/classes/GlobalScreen/class.ilLSCompletionNotificationProvider.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\GlobalScreen\Scope\Notification\Provider\AbstractNotificationProvider;
+use ILIAS\Notifications\Repository\ilNotificationOSDRepository;
+use ILIAS\UI\Component\Symbol\Icon\Standard;
+use ILIAS\Notifications\ilNotificationOSDHandler;
+
+class ilLSCompletionNotificationProvider extends AbstractNotificationProvider
+{
+    public const NOTIFICATION_TYPE = 'lso_completion';
+    public const NOTIFICATION_TIME_PREFERENCE_KEY = 'lso_completion_note_ts';
+
+    public function getNotifications(): array
+    {
+        $current_user = $this->dic['ilUser'];
```

Comment:
I would strongly recommend to use the access function instead of acces dic properties by array keys 
(just like `$this->dic->user()`)

Pls apply that for all array access calls on the DIC

Reply:
Why is this? I usually switch it the other way around, since those function are mere wrappers and somewhat artificial?

Reply:
Basiacly the use of wrapper is heavly recommended. Not only this provides a general API on wich the crowd of devleopers can base their inner dependencies, it also help in programming since hard wrapper allow autocompletions. The array access is a wrapper itsself with the exception that this is a dynamicly provided wrapper withing the construct of the PHP array. Therefore nothing is safed if you acces it by the array key, we only losing control of the wrapper on this point.

Majorly it for the same reason we dont use `$_GET` anymore but the `$DIC->http()->request()->getQueryParams()` since it gives us controll over the object structure and improves maintainability and extensibility of code in general (Not only ILIAS) due to developer handlet access wich can be regulated and swapped out.

There is also a small paragaprh inside the DI Docu about this https://github.com/ILIAS-eLearning/ILIAS/blob/release_8/src/DI/README.md